### PR TITLE
cli: gather raw option

### PR DESCRIPTION
**Pull request by richardjgowers**
fixes issue #557 
fixes issue #618

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenFreeEnergy/openfe/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

### CODE REVIEW

**Comment by richardjgowers** [comment]
I've not added tests for this as the current test files don't have the correct info (see #605 and #595 for ongoing context)

This is what running this patch locally produces: 
```
leg     ligand_i        ligand_j        DG(i->j) (kcal/mol)     uncertainty (kcal/mol)
complex lig_ejm_31      lig_ejm_42      -14.77  0.04
complex lig_ejm_31      lig_ejm_42      -14.74  0.04
complex lig_ejm_31      lig_ejm_42      -14.94  0.04
solvent lig_ejm_31      lig_ejm_42      -15.68  0.03
solvent lig_ejm_31      lig_ejm_42      -15.69  0.03
solvent lig_ejm_31      lig_ejm_42      -15.64  0.03
complex lig_ejm_31      lig_ejm_46      -40.56  0.06
complex lig_ejm_31      lig_ejm_46      -40.76  0.05
complex lig_ejm_31      lig_ejm_46      -40.90  0.04
solvent lig_ejm_31      lig_ejm_46      -39.92  0.04
solvent lig_ejm_31      lig_ejm_46      -39.94  0.04
solvent lig_ejm_31      lig_ejm_46      -39.95  0.04
complex lig_ejm_31      lig_ejm_47      -27.68  0.08
complex lig_ejm_31      lig_ejm_47      -27.80  0.06
complex lig_ejm_31      lig_ejm_47      -27.51  0.07
solvent lig_ejm_31      lig_ejm_47      -27.83  0.05
solvent lig_ejm_31      lig_ejm_47      -27.84  0.05
solvent lig_ejm_31      lig_ejm_47      -27.88  0.05
complex lig_ejm_31      lig_ejm_48      -16.15  0.08
complex lig_ejm_31      lig_ejm_48      -15.96  0.07
complex lig_ejm_31      lig_ejm_48      -16.01  0.08
solvent lig_ejm_31      lig_ejm_48      -16.83  0.06
solvent lig_ejm_31      lig_ejm_48      -16.65  0.07
solvent lig_ejm_31      lig_ejm_48      -16.77  0.06
complex lig_ejm_31      lig_ejm_50      -57.31  0.04
complex lig_ejm_31      lig_ejm_50      -57.45  0.04
complex lig_ejm_31      lig_ejm_50      -57.37  0.04
solvent lig_ejm_31      lig_ejm_50      -58.33  0.04
solvent lig_ejm_31      lig_ejm_50      -58.42  0.04
solvent lig_ejm_31      lig_ejm_50      -58.19  0.04
complex lig_ejm_42      lig_ejm_43      -19.24  0.04
complex lig_ejm_42      lig_ejm_43      -18.72  0.05
complex lig_ejm_42      lig_ejm_43      -18.94  0.04
solvent lig_ejm_42      lig_ejm_43      -20.17  0.03
solvent lig_ejm_42      lig_ejm_43      -20.28  0.03
solvent lig_ejm_42      lig_ejm_43      -20.23  0.03
complex lig_ejm_46      lig_jmc_23      17.31   0.02
complex lig_ejm_46      lig_jmc_23      17.37   0.02
complex lig_ejm_46      lig_jmc_23      17.35   0.02
solvent lig_ejm_46      lig_jmc_23      17.20   0.02
solvent lig_ejm_46      lig_jmc_23      17.40   0.02
solvent lig_ejm_46      lig_jmc_23      17.30   0.02
complex lig_ejm_46      lig_jmc_27      15.84   0.03
complex lig_ejm_46      lig_jmc_27      15.79   0.03
complex lig_ejm_46      lig_jmc_27      15.80   0.03
solvent lig_ejm_46      lig_jmc_27      16.16   0.03
solvent lig_ejm_46      lig_jmc_27      16.01   0.03
solvent lig_ejm_46      lig_jmc_27      16.07   0.03
complex lig_ejm_46      lig_jmc_28      23.43   0.04
complex lig_ejm_46      lig_jmc_28      23.29   0.04
complex lig_ejm_46      lig_jmc_28      23.17   0.04
solvent lig_ejm_46      lig_jmc_28      23.67   0.03
solvent lig_ejm_46      lig_jmc_28      23.61   0.03
solvent lig_ejm_46      lig_jmc_28      23.65   0.03

```

File: openfecli/tests/commands/test_gather.py [hunk 1]
```diff
@@ -145,3 +205,29 @@ def test_missing_leg_allow_partial(results_dir):
...
+@pytest.fixture
+def rbfe_results():
+    # fetches rbfe results from online
+    # untars into local directory and returns path to this
+    d = RBFE_RESULTS.fetch('results.tar.gz', processor=pooch.Untar())
+
+    return os.path.join(pooch.os_cache('openfe'), 'results.tar.gz.untar', 'results')
+
+
+@pytest.mark.download
```
Comment (richardjgowers) [inline openfecli/tests/commands/test_gather.py:226]: we should probably also add in a pytest plugin that turns tests requiring downloads off by default?

File: openfecli/commands/gather.py [hunk 2]
```diff
@@ -344,6 +370,7 @@ def gather(rootdir, output, report, allow_partial):
         'dg': _write_dg_mle,
         'ddg': _write_ddg,
         'dg-raw': _write_dg_raw,
+        'raw': _write_raw,
```
Comment (IAlibay) [inline openfecli/commands/gather.py:373]: This is a really confusing option - we now have two very similar methods that match the exact same naming.

"raw" gives you raw dGs for all individual replicas
"dg-raw" gives you average dGs and stdevs across all replicas

My take here (and in #558) is that "dg-raw" isn't "raw", it's already processed. I'm not going to stand here and say "we need to change everything", but we need a better way to clarify this to users. Right now if I read the docstring I get two different options that start with "X reports the raw results,".

**Review by IAlibay — COMMENTED** [review]
I think we need to discuss what "raw" means, or at least how we're not going to confuse users.

I'm not going to ask to write much more, but we should at least consider if we should write better docs?

Pinging @hannahbaumann for thoughts here too.

File: openfecli/commands/gather.py [hunk 3]
```diff
@@ -171,24 +180,35 @@ def _get_ddgs(legs, error_on_missing=True):
...
-            DDGhyd, hyd_unc = format_estimate_uncertainty(DDGbind,
-                                                          bind_unc)
+            DDGhyd, hyd_unc = format_estimate_uncertainty(DDGhyd, hyd_unc)
             writer.writerow([ligA, ligB, DDGhyd, hyd_unc])
 
 
-def _write_dg_raw(legs, writer, allow_partial):
+def _write_raw(legs, writer, allow_partial=True):
+    writer.writerow(["leg", "ligand_i", "ligand_j", "DG(i->j) (kcal/mol)",
+                     "uncertainty (kcal/mol)"])
```
Comment (IAlibay) [inline openfecli/commands/gather.py:195]:
```suggestion
                     "MBAR uncertainty (kcal/mol)"])
```
sorry just a slight precision

**Review by IAlibay — APPROVED** [review]
@richardjgowers just the one thing, other LGTM (of course ignoring all the failing tests)

**Pull request merged**